Add Gemini Vision interactive CLI tool for image analysis

emredeveloper opened the pull request:
- Create interactive CLI tool using smolagents and Gemini Vision API
- Add image analysis, code extraction, and comparison features
- Include documentation and requirements
- Implement user-friendly command interface
- Support code detection and automatic file saving

emredeveloper committed:
Add Gemini Vision interactive CLI tool for image analysis

- Create interactive CLI tool using smolagents and Gemini Vision API
- Add image analysis, code extraction, and comparison features
- Include documentation and requirements
- Implement user-friendly command interface
- Support code detection and automatic file saving

emredeveloper committed:
Add enhanced Gemini Vision analysis tool with focused agent functionality

- Implement focused CodeAgent with strict system instructions
- Add screenshot capture and analysis capabilities
- Improve tool interaction with more deterministic behaviors
- Fix issue where tools would perform unintended additional actions
- Add auto-correction for common command syntax
- Update documentation with improved usage examples
- Support direct function calling in CodeAgent mode

emredeveloper committed:
update

aymeric-roucher commented on examples/gemini_vision_agent/gemini_vision_agent.py:
```diff
@@ -0,0 +1,615 @@
+        analyze_screenshot
+    ]
+    
+    # Özel bir sistem prompt'u tanımla
```
Some Turkish comments here and there: please translate them to English!

emredeveloper replied:
I did

emredeveloper committed:
feat: Translate Turkish comments to English

This commit translates all Turkish comments in the gemini_vision_agent.py file to English to maintain consistent documentation and improve code readability for international developers.

Changes include:
- Translating temperature reduction comment
- Translating system prompt definition comment
- Translating max_steps adjustment comment
- Translating base_tools configuration comment
- Translating system_prompt parameter comment

The code functionality remains unchanged; this is purely a documentation improvement.

aymeric-roucher commented on examples/gemini_vision_agent/gemini_vision_agent.py:
```diff
@@ -0,0 +1,615 @@
+        verbosity_level=LogLevel.INFO,
+        max_steps=7,  # Reduce processing steps
+        add_base_tools=False,  # Disable basic tools to focus on vision capabilities
+        system_prompt=system_prompt  # Add custom system instructions
```
Did you test the agent with latest version?

emredeveloper replied:
Yes, it was built on the latest version v1.12.0, and it didn't give any warnings or problems when running

emredeveloper replied:
I've missed applying some of the tests; I'll fix them and get back to you with feedback

emredeveloper replied:
make quality      
ruff check examples src tests utils
All checks passed!
ruff format --check examples src tests utils
64 files already formatted
python utils/check_tests_in_ci.py
✅ All good!

emredeveloper committed:
Make test :
gemini_vision_agent.py code is not the source of these issues.

emredeveloper committed:
fix: format gemini_vision_agent.py to meet project code style

This commit fixes formatting issues in the Gemini Vision Agent example to follow
the smolagents code style guidelines. The following changes were made:

- Fixed whitespace in blank lines within docstrings
- Applied proper quotation style (double quotes instead of single quotes)
- Adjusted spacing around operators and commas
- Improved indentation consistency
- Added trailing comma in multi-line collections
- Fixed line breaks according to ruff formatting rules

All quality checks are now passing: ruff check and ruff format.

emredeveloper committed:
Merge branch 'huggingface:main' into main

aymeric-roucher commented:
@emredeveloper please test your PR with the latest version of smolagents. In the current state it's clear you didn't, i'll let you try and find out why 😉

emredeveloper committed:
Update Gemini Vision Agent example for smolagents 1.13.0.dev0

- Modified CodeAgent initialization to be compatible with smolagents 1.13.0.dev0
- Removed system_prompt parameter and used description instead
- Fixed test suite by properly mocking dependencies
- Ensured display_image test passes by using effective mocking
- Fixed test_create_smolagent to work with the current codebase
- Optimized imports and error handling
- Updated requirements.txt to use smolagents version 1.12.0

emredeveloper committed:
Update Gemini Vision Agent example for smolagents 1.13.0.dev0

- Modified CodeAgent initialization to be compatible with smolagents 1.13.0.dev0
- Removed system_prompt parameter and used description instead
- Fixed test suite by properly mocking dependencies
- Ensured display_image test passes by using effective mocking
- Fixed test_create_smolagent to work with the current codebase
- Optimized imports and error handling
- Updated requirements.txt to use smolagents version 1.12.0

emredeveloper committed:
Merge branch 'main' of https://github.com/emredeveloper/smolagents

emredeveloper commented:
@aymeric-roucher I think I've understood and solved it, hopefully, I'm not wrong, haha

aymeric-roucher commented:
@emredeveloper that's the direction! You can follow this guide to fix system prompt: https://huggingface.co/docs/smolagents/tutorials/building_good_agents#3-change-the-system-prompt-generally-not-advised

emredeveloper commented:
@aymeric-roucher okay...

emredeveloper committed:
Merge branch 'huggingface:main' into main

emredeveloper commented:
Thanks for the feedback and guidance. This PR has become stale, and its current approach is no longer aligned with the recommended native VLM setup. I’m closing it to keep the contribution queue clean.
